rt/rt.mjs

```diff
@@ -238,29 +272,40 @@ export class Engine {
             }
             const tag = extract_tag(ptr);
             switch (tag) {
-                case TAGS.fixnum: return ptr;
-                case TAGS.constant: return ptr;
-                case TAGS.pair: {
-                    let car = from_space[extract_value(ptr) * 2];
-                    let cdr = from_space[extract_value(ptr) * 2 + 1];
-                    car = deep_copy(car);
-                    cdr = deep_copy(cdr);
-                    const p = alloc(2);
-                    mem_i32[p / 4] = car;
-                    mem_i32[p / 4 + 1] = cdr;
-                    const new_ptr = replace_tag(p, TAGS.pair);
-                    forwards.set(ptr, new_ptr);
-                    return new_ptr;
-                }
-                case TAGS.character: return ptr;
-                case TAGS.string: {
+            case TAGS.fixnum: return ptr;
+            case TAGS.constant: return ptr;
+            case TAGS.pair: {
+                let car = from_space[extract_value(ptr) * 2];
+                let cdr = from_space[extract_value(ptr) * 2 + 1];
+                car = deep_copy(car);
+                cdr = deep_copy(cdr);
+                const p = alloc(2);
+                mem_i32[p / 4] = car;
+                mem_i32[p / 4 + 1] = cdr;
+                const new_ptr = replace_tag(p, TAGS.pair);
+                forwards.set(ptr, new_ptr);
+                return new_ptr;
+            }
+            case TAGS.character: return ptr;
+            case TAGS.string: {
+                if (extract_tag(from_space[extract_value(ptr) * 2]) == TAGS.fixnum) {
+                    // new style string
```

Comment:
Also needs follow-up to remove old case.  This case should change to be TAGS.vector though, AFAIU.